src/block_compressor.rs

```diff
@@ -89,8 +89,20 @@ impl GpuBlockCompressor {
             device.create_shader_module(include_wgsl!("shader/bc1_to_5.wgsl"));
         #[cfg(feature = "bc6h")]
         let shader_module_bc6h = device.create_shader_module(include_wgsl!("shader/bc6h.wgsl"));
+        // Temporary fix - WGPU is converting the loop from shader code
+        // in a weird way on my machine
+        // Bug introduced in https://github.com/gfx-rs/wgpu/pull/7080
         #[cfg(feature = "bc7")]
-        let shader_module_bc7 = device.create_shader_module(include_wgsl!("shader/bc7.wgsl"));
+        let shader_module_bc7 = {
+            let checks = ShaderRuntimeChecks {
+                bounds_checks: true,
+                force_loop_bounding: false,
+            };
+            let module = unsafe {
```

Comment:
You can remove the module variable and return it directly. Maybe you can move tne unsafe itself to the outer most block.